docs/upgrade-guide.md

```diff
@@ -36,6 +36,13 @@
 - `PerspectiveView` class - use `FirstPersonView`
 - `ThirdPersonView` class - use `MapView` (geospatial) or `OrbitView` (info-vis)
 
+##### Debugging
+
+Type `deck.debug(true)` in the console together with `deck.log.priority=<number>` to enable debug mode.
```

Comment:
I'm thinking `deck.debug(true)` should set a minimum log priority to get logging so this can be done in one step (with the user having the option to set the priority higher). And maybe the log priority could be an optional second argument to `deck.debug`.

Reply:
> I'm thinking deck.debug(true) should set a minimum log priority to get logging

Agree that we should make this easy for user. This logging system has always had a discovery problem - how many deck.gl users are aware of it and able to leverage it?

For reference, the probe.gl `Log.enable()` etc returns `this` for chaining. `deck.log.enable().setPriority(3)` etc.  It is a reasonable design, but the user needs to know that those methods exist.

Reply:
An easy solution to this is to set the default `priority` of `deck.log` to a non-zero number.

If we want to address this usability issue in probe.gl, I would set the priority default to `1`. When log is disabled only `priority=0` messages will pass through. Since `enabled` is default to `false`, the logging behavior will remain the same for most users who haven't messed with it.

Reply:
> An easy solution to this is to set the default priority of deck.log to a non-zero number. 

Yes. The system is probably not optimal. I would not be opposed to changes. They could be small as you suggest but here is some additional grist for the mill:

- We have both enable and priority, maybe we don't need both? Initial probe.gl users (a long time ago) had concerns about perf impact when not in use, and the idea was that `enable(false)` would replace all methods with empty functions, not a bad idea though that feature generated surprising amounts of errors during probe.gl maintenance, not sure it was worth it. 

Another question is whether `enable/disable` should affect `log.warn` and `log.error` as we now rely on those for non-debugging cases. If `disable` does not defeat those, it does not really so what it sounds like, and if it does, `enable` should probably be true by default as I assume we do want such messages...

Another criticism on the probe.gl API that was often raised: The term `priority` was considered very confusing since higher values are less likely to be logged. Most logging framework calls this value "log level", and it would make perfect sense to align with that convention.